staging/src/k8s.io/client-go/rest/request_watchlist_test.go

```diff
@@ -0,0 +1,337 @@
+		},
+		{
+			name:        "an err is always returned",
+			result:      nil,
+			target:      WatchListResult{err: fmt.Errorf("dummy err")},
+			expectedErr: fmt.Errorf("dummy err"),
+		},
+		{
+			name:   "empty list",
+			result: &v1.PodList{},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "gv is applied",
+			result: &v1.PodList{},
+			target: WatchListResult{gv: schema.GroupVersion{Group: "g", Version: "v"}},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList", APIVersion: "g/v"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "gv is applied, empty group",
+			result: &v1.PodList{},
+			target: WatchListResult{gv: schema.GroupVersion{Version: "v"}},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList", APIVersion: "v"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "rv is applied",
+			result: &v1.PodList{},
+			target: WatchListResult{initialEventsEndBookmarkRV: "100"},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList"},
+				ListMeta: metav1.ListMeta{ResourceVersion: "100"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "items are applied",
+			result: &v1.PodList{},
+			target: WatchListResult{items: []runtime.Object{makePod(1), makePod(2)}},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList"},
+				Items:    []v1.Pod{*makePod(1), *makePod(2)},
+			},
+		},
+		{
+			name:        "type mismatch",
+			result:      &v1.PodList{},
+			target:      WatchListResult{items: []runtime.Object{makeNamespace("1")}},
+			expectedErr: fmt.Errorf("received object type = v1.Namespace at index = 0, doesn't match the list item type = v1.Pod"),
+		},
+	}
+	for _, scenario := range scenarios {
+		t.Run(scenario.name, func(t *testing.T) {
+			err := scenario.target.Into(scenario.result)
+			if scenario.expectedErr != nil && err == nil {
+				t.Fatal("expected an error")
+			}
+			if scenario.expectedErr == nil && err != nil {
+				t.Fatal("didn't expect an error")
+			}
+			if err != nil {
+				if scenario.expectedErr.Error() != err.Error() {
+					t.Fatalf("unexpected err = %v, expected = %v", err, scenario.expectedErr)
+				}
+				return
+			}
+			if !apiequality.Semantic.DeepEqual(scenario.expectedResult, scenario.result) {
+				t.Errorf("diff: %v", cmp.Diff(scenario.expectedResult, scenario.result))
+			}
+		})
+	}
+}
+
+func TestWatchListSuccess(t *testing.T) {
+	scenarios := []struct {
+		name           string
+		watchEvents    []watch.Event
+		expectedResult *v1.PodList
+	}{
+		{
+			name: "happy path",
+			watchEvents: []watch.Event{
+				{Type: watch.Added, Object: makePod(1)},
+				{Type: watch.Added, Object: makePod(2)},
+				{Type: watch.Bookmark, Object: makeBookmarkEvent(5)},
+			},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core/v1",
```

Comment:
API version for core api group is "v1" (not core/v1)

[same below]

Reply:
done.